modules/shortcodes/email-subscribe.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * WARNING: This file is distributed verbatim in Jetpack.
+ * There should be nothing WordPress.com specific in this file.
+ *
+ * @hide-in-jetpack
+ **/
+
+/**
+ * Class Jetpack_Email_Subscribe
+ * This class encapsulates shortcode for subscribing to a MailChimp list.
+ * It displays a simple signup form that gets an email address from the user and signs him for a list
+ * selected in "Sharing" section in calypso.
+ * Other Email services can be implemented as well in the future.
+ */
+class Jetpack_Email_Subscribe {
+
+	private static $shortcode = 'jetpack-email-subscribe';
```

Comment:
Should the shortcode be more Mailchimp specific? E.g. `jetpack-mailchimp-subscribe`? This seems pretty generic — unless it's in purpose and the actual backend service could be switched over to anything else in future?

Reply:
>  it's in purpose and the actual backend service could be switched over to anything else in future?

That is precisely the case.